### DISCUSSION TITLE
machine reconcile more frequently than expected.

### DISCUSSION THREAD

**Discussion by jianhang33** [kubernetes-sigs/cluster-api#10862] [Q&A] [answered]
In my machine's Reconcile function, I have
```
  return ctrl.Result{RequeueAfter: 60 * time.Second}, nil
```
I expect machine reconciliation happens once every 60 seconds, but it actually happens three times a second, for a single machine. It seems that the request is enqueued immediately, rather than in 60 seconds.

The only place I can see that might enqueue the request is in MyMachine.SetupWithManager where we add two watches.
```
    ctrl.NewControllerManagedBy(mgr).
      For(&infrav1.MyMachine{}).
      Watches(
          &clusterv1.Machine{},                    
          handler.EnqueueRequestsFromMapFunc(util.MachineToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("MyMachine"))),
        ).
	Watches(
	    &infrav1.MyCluster{},
	    handler.EnqueueRequestsFromMapFunc(r.MyClusterToMyMachines(ctx)),
	).
	Build(r)
```
but I don't know how to validate it. Is there a way for me to see who enqueues reconciliation requests? Any help/suggestion is appreciated.

**Comment by sbueringer** [comment]
You get reconcile requests whenever mymachine, machine or mycluster update events occur.

You can add logging to the map funcs if you want

**Reply by jianhang33** [reply]
Thanks. It seemed that the cluster was updated because the function returned by r.MyClusterToMyMachines(ctx) was called frequently, however when I check the events, I don't see that many
```
kubectl get events --sort-by='.lastTimestamp' -n ns
```
For example, the last three events were seen at 3m30s, 3m29s, 3m28s. Is there other way that triggers reconcile?

**Reply by sbueringer** [reply]
The controller is not watching on these kind of events. there are "watch events" sent whenever an object is created/updated/deleted

**Reply by jianhang33** [reply, marked as answer]
Got it. I can see continuous output from
```
kubectl get MyCluster -n ns --watch
```
Thank you very much @sbueringer.
